models/org_team.go

```diff
@@ -511,6 +546,14 @@ func UpdateTeam(t *Team, authChanged bool) (err error) {
 		}
 	}
 
+	// Add all repositories to the team if it has access to all of them.
+	if includeAllChanged && t.IncludesAllRepositories {
```

Comment:
includeAllChanged is just for optimization, we don't need to redo this every time someone change settings of a team that has all repositories. If include all was changed to not include all repositories, we don't need to add them.

Reply:
I mean, if the admin changes from "_All Repositories_" into "_Specific Repositories_", are we leaving all the repos the team had or should we remove them? I think we should remove them.

Reply:
We can't restore the team repositories to before setting all repositories. As a user I don't think it would be expected to remove all repositories when changing to "specific repositories". I see the use of being able to remove all team repositories but I think it shall be a separate button (on the team repository tab)

Reply:
imho removing should be done manually

Reply:
> 
> 
> imho removing should be done manually

:worried: Doing it one by one can be 1) error prone (which is not good for security procedures) and 2) very taxing.

Some power users may have the possibility of going through SQL for this (as suggested earlier), but I'm not a fan of asking the user to know SQL, Gitea architecture, etc., especially when there's no document for them to go for reference. And the org owner might not have admin permissions or access to the server after all.

I think a checkbox to remove automatically is very easy to implement and should not add much clutter.

Reply:
I don't understand why removing one by one is error prone but cumbersome, yes. 

I thought of something like this (with warning):
![image](https://user-images.githubusercontent.com/161914/68161220-b1b14a80-ff55-11e9-968f-ac6fbee3283c.png)

But prefer to do it in a different PR.

Reply:
From the screenshot you've pasted I realize that the situation is not as complex as I pictured it. To remove the repos from the team the owner only needs to go to the team repos tab (which is cumbersome, but not terrible). My only concern is that they forget to do that, but the warning messages should be enough.